### ISSUE TITLE
Schedule live stream from mp4

### ISSUE THREAD

**Issue by kisso** [kaltura/nginx-vod-module#1510]
Hi, is it possible to schedule live stream at specific time using nginx-vod-module? I need this stream to be not rewindable. I want to embed link to playlist at my website, and for example need to start stream at 8PM.

**Comment by erankor** [comment]
Yes, it's possible, you need to use mapped mode with a JSON like -
```
{
	"playlistType": "live",
	"discontinuity": true,
	"firstClipTime": 1451918170000,
	"durations": [1800000],
	"sequences": [
		{
			"clips": [
				{
					"type": "source",
					"path": "/path/to/video1.mp4"
				}
			]
		}
	]
}
```
You need to set firstClipTime to the broadcast start time, in unix-time millis. If the stream is loaded outside the broadcast time, it will return an error. So it may be a good idea to stitch some slate before + after the main video, either by including them in the JSON, or by concatenating them to the source MP4.
Regarding the ability to rewind, you can use `vod_live_window_duration` to the control the number of seconds of video returned in the manifest. Note that it is not recommended to set this value below 3 segments.

**Comment by kisso** [comment]
Okay thank you, and what about discontinuty why should I enable it?

**Comment by erankor** [comment]
With one video, it's irrelevant, doesn't matter what you put there... 
If you will embed a slate before/after the video, you will most likely need to set it to true, unless you take extra care to ensure the videos have identical SPS/PPS.

**Comment by kisso** [comment]
And what about presentationEndTime or expiredTime? Should I used them? I tried presentationEndTime and the video continued after specified time

**Comment by erankor** [comment]
These params provide more fine grained control, intended mostly for "real" live streams. `presentationEndTime` controls whether the module returns `EXT-X-ENDLIST`,  `expirationTime` can be used to make the module return 404 after a certain time. 
I don't think you need them here, as far as I remember, the module will return an error, when the clock moves out of the window defined in the JSON.

**Comment by kisso** [comment]
Super thank you so much, also I am able to use multibitrate stream right?

**Comment by erankor** [comment]
Yup, you just need to add more elements to the `sequences` array

**Comment by kisso** [comment]
I tried and getting error: ngx_http_vod_validate_streams: invalid sequence count 3 while sending to client

**Comment by erankor** [comment]
Are you requesting `index.m3u8`? 
You can't request `index.m3u8` when there are multiple sequences, you need to use `master.m3u8`

**Comment by kisso** [comment]
Yes i am requesting master.m3u8

**Comment by kisso** [comment]
I have another problem, when I use only one sequence and requesting master.m3u8 I am getting whole video which I can rewind, when I request on index.m3u8 I am getting livestream is this correct behavior? Anybody can change url from index to master and see whole video, I don't want this

**Comment by erankor** [comment]
this doesn't make sense... the master.m3u8 references the index.m3u8 internally, there should be no difference between the two. maybe you have some caching issue.
